File: backend/app/schemas/data_deletion.py
from pydantic import BaseModel, EmailStr, Field, model_validator


class DataDeletionRequest(BaseModel):
    parent_name: str = Field(min_length=1, max_length=120)
    email: EmailStr
    child_name: str | None = Field(default=None, max_length=120)
    request_details: str | None = Field(default=None, max_length=1200)
    confirmation_accepted: bool = False

    @model_validator(mode='after')
    def validate_confirmation(self):
        if not self.confirmation_accepted:
            raise ValueError('Please confirm that this request will be reviewed before deletion.')
        return self


class DataDeletionResponse(BaseModel):
    success: bool = True
    message: str = 'Your deletion request has been received. We will review it and contact you if needed.'

File: frontend/src/lib/api/dataDeletion.ts
import { apiPost } from '../api';

export type DataDeletionPayload = {
  parent_name: string;
  email: string;
  child_name?: string;
  request_details?: string;
  confirmation_accepted: boolean;
};

export type DataDeletionResponse = {
  success: boolean;
  message: string;
};

export function submitDataDeletionRequest(payload: DataDeletionPayload, accessToken?: string): Promise<DataDeletionResponse> {
  const headers = accessToken ? { Authorization: `Bearer ${accessToken}` } : undefined;
  return apiPost<DataDeletionResponse>('/api/data-deletion/request', payload, headers);
}
